zerver/tornado/event_queue.py

```diff
@@ -1075,6 +1092,17 @@ def get_client_payload(apply_markdown: bool, client_gravatar: bool) -> Dict[str,
     for user_data in users:
         user_profile_id: int = user_data["id"]
         flags: Collection[str] = user_data.get("flags", [])
+
+        # TODO/compatibility: Translation code to replace the deprecated
+        # `wildcard_mentioned` flag with `stream_wildcard_mentioned`.
+        # Remove this when one can no longer directly upgrade from 7.x to main.
+        if "wildcard_mentioned" in flags:
+            updated_flags = [
+                "stream_wildcard_mentioned" if flag == "wildcard_mentioned" else flag
+                for flag in flags
+            ]
+            flags = updated_flags
```

Comment:
Hmm, I don't think we need this sort of compatibility code. My thinking is that:
* An event that is emitted by a post-upgrade server will have all 3 fields, so this doesn't apply. (And a client that had started with a pre-8.0 server will probably just ignore the new fields, and happily access `wildcard_mentioned` until it restarts and gets an 8.0 feature level startup)
* An event that is emitted by a pre-upgrade server will just have a `wildcard_mentioned` field, and will only ever be processed by a client that had created its event queue expecting a pre-8.0 server and thus should be OK processing an additional pre-8.0 format event even if it's also getting post-8.0 events now.

Reply:
Yeah, passing the three fields via send_event, instead of including `wildcard_mentioned` in tornado code is a much cleaner approach.